Podfile

```diff
@@ -47,7 +47,7 @@ def wordpress_ui
 end
 
 def wordpress_kit
-  pod 'WordPressKit', '~> 4.58.1-beta.1'
+  pod 'WordPressKit', '~> 4.58', '>= 4.58.1'
```

Comment:
Any reason for that change?

Why not keep the `~>` notation, to at least avoid allowing an hypothetical `5.x.x` version to break things?

Moreover, why not just keep `~> 4.58` then use `bundle update WordPressKit` to update the used version to `4.58.1` in the lockfile while still respect and allow semantic versioning? `>= 4.58.1` feels too loose and open to breaking changes imho

Reply:
> Why not keep the ~> notation

@AliSoftware from the way your question is worded, it seems like my diff removed the `~> 4.58` requirement. However, it did not. It adds `>= 4.58.1` to it. 

My rationale for adding `>= 4.58.1` is to ensure that we always get _at least_ `4.58.1`. I did that because I think using `~> 4.58` could potentially result in a resolution of `4.58.0` is another dependency has a hard requirement on that version.

(We need `4.58.1` because of a bugfix).

`'~> 4.58', '>= 4.58.1'` will avoid allowing an hypothetical `5.x.y` version to break things. I checked this by using `'~> 1.5', '>= 1.5.1'`, which resolved to `1.8.0` in https://github.com/wordpress-mobile/WordPress-iOS/commits/5db3cf6fe3be33d7346ed6a21e03c63e57a3a189 .

Does this address your concern? 

---

Assuming I got the configuration right, what I like about it is the granularity and how it specifies to a reader that there's something "special" about `4.58.1` in relationship to the code.

At the same time,  I can see this being confusing for people who don't spend much time tinkering with dependency managers. 🤔 A single `~> MAJOR.MINOR` would definitely be simpler. And, at the end of the day, we don't have many internal libraries and transitive dependencies at this point, and we might be well served by a simpler setup.

Reply:
> from the way your question is worded, it seems like my diff removed the ~> 4.58 requirement. However, it did not. It adds >= 4.58.1 to it.

Oh, good point, I totally misread that diff! ☕  Thanks for the precision 🙂 

In that case it all makes sense 👍

Reply:
Yep, totally :) Good thinking 👍 
(I wasn't suggesting to do `~> 4.58.1`, but just to keep `~> 4.58` and rely on `pod update` to pick `4.58.1`… but now that I see that your `>= 4.58.1` is an addition not a replacement, this all fits and covers all the cases we want indeed 👍